src/app/wallet/components/crypto-list/crypto-list.component.ts

```diff
@@ -1145,7 +1145,11 @@ export class CryptoListComponent implements OnInit, OnDestroy {
       window.open(this.etherscanUrl + tokenAddress, '_blank');
     }
   }
+  goToCryptoDetails(crypto : any){
+    const cryptoUpperCase = crypto.toUpperCase();
+    this.router.navigate(['/wallet/coin-detail'], { queryParams: { crypto: cryptoUpperCase } });
```

Comment:
Error Handling: There's currently no error handling in case navigation fails for some reason (like if this.router is undefined). Adding some error handling could improve the robustness of your code.

Code Commenting: Adding some comments to describe the purpose of this function and why you're converting crypto to uppercase would be beneficial for other developers reading your code.